File: an2_sem4/MPP/Homework/sapt9/java/service/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/service/IAppService.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Angajat;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.EntityRepoException;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service.observer.Observable;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service.observer.Observer;

import java.time.LocalDateTime;

public interface IAppService {
    Angajat loginAngajat(String username, String password, Observer client) throws ServiceException;
    void registerAngajat(String username, String password, String email) throws ServiceException;

    Iterable<Spectacol> getAllSpectacole() throws ServiceException;

    Iterable<Spectacol> filterSpectacole(LocalDateTime startDate, LocalDateTime endDate) throws ServiceException;

    Iterable<Spectacol> filterSpectacole(LocalDateTime day) throws ServiceException;

    void reserveBilet(Spectacol spectacol, String cumparatorName, int seats) throws BiletReservationException, ServiceException;

    void logout(Angajat angajat);
}

File: an2_sem4/MPP/Homework/sapt56/cs/FestivalSellpoint/FestivalSellpoint.Service/IAppService.cs
using FestivalSellpoint.Domain;
using System.Collections.Generic;
using System;

namespace FestivalSellpoint.Service
{
    public interface IAppService
    {
        Angajat LoginAngajat(string username, string password);
        void RegisterAngajat(string username, string password, string email);


        IEnumerable<Spectacol> GetAllSpectacole();

        IEnumerable<Spectacol> FilterSpectacole(DateTime startDate, DateTime endDate);

        IEnumerable<Spectacol> FilterSpectacole(DateTime day);

        void ReserveBilet(Spectacol spectacol, string cumparatorName, int seats);
    }
}
